hedera-node/hedera-app/src/xtest/java/contract/HbarApproveXTest.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package contract;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_ALLOWANCE_SPENDER_ID;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+import static contract.HtsErc721TransferXTestConstants.APPROVED_HEADLONG_ADDRESS;
+import static contract.HtsErc721TransferXTestConstants.UNAUTHORIZED_SPENDER_ADDRESS;
+import static contract.HtsErc721TransferXTestConstants.UNAUTHORIZED_SPENDER_HEADLONG_ADDRESS;
+import static contract.HtsErc721TransferXTestConstants.UNAUTHORIZED_SPENDER_ID;
+import static contract.XTestConstants.AN_ED25519_KEY;
+import static contract.XTestConstants.OWNER_ADDRESS;
+import static contract.XTestConstants.OWNER_BESU_ADDRESS;
+import static contract.XTestConstants.OWNER_HEADLONG_ADDRESS;
+import static contract.XTestConstants.OWNER_ID;
+import static contract.XTestConstants.SENDER_CONTRACT_ID_KEY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.esaulpaugh.headlong.abi.Tuple;
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.state.primitives.ProtoBytes;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.has.hbarapprove.HbarApproveTranslator;
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.tuweni.bytes.Bytes;
+
+/**
+ * Exercises hbar allowance for a token via the following steps relative to an {@code OWNER} and {@code SENDER} accounts:
+ * <ol>
+ *     <li>Set Hbar Approval of OWNER for SPENDER via {@link HbarApproveTranslator#HBAR_APPROVE_PROXY}.</li>
+ *     <li>Set Hbar Approval of OWNER for SPENDER via {@link HbarApproveTranslator#HBAR_APPROVE}.</li>
+ *     <li>Fail Hbar Allowance if SPENDER is not found.</li>
+ *  * </ol>
+ */
+public class HbarApproveXTest extends AbstractContractXTest {
+    @Override
+    protected void doScenarioOperations() {
+        runHasCallAndExpectOnSuccess(
+                OWNER_BESU_ADDRESS,
+                bytesForRedirectForAccount(
+                        HbarApproveTranslator.HBAR_APPROVE_PROXY.encodeCallWithArgs(
+                                UNAUTHORIZED_SPENDER_HEADLONG_ADDRESS, BigInteger.valueOf(100L)),
+                        OWNER_ID),
+                output -> assertEquals(
+                        asBytesResult(
+                                HbarApproveTranslator.HBAR_APPROVE.getOutputs().encodeElements((long)
+                                        SUCCESS.getNumber())),
+                        output),
+                "Successful execution of hbarAllowance for SENDER by OWNER.");
```

Comment:
Here and at L75 `hbarAllowance` -> `hbarApprove`, plus should be different text?

Reply:
Still needs a fix if you want